dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/HadoopUtils.java

```diff
@@ -421,15 +418,15 @@ public ExecutionStatus getApplicationStatus(String applicationId) throws JSONExc
 
         String responseContent = HttpUtils.get(applicationUrl);
         if (responseContent != null) {
-            JSONObject jsonObject = JSON.parseObject(responseContent);
-            result = jsonObject.getJSONObject("app").getString("finalStatus");
+            ObjectNode jsonObject = JSONUtils.parseObject(responseContent);
+            result = jsonObject.path("app").path("finalStatus").asText();
         } else {
             //may be in job history
             String jobHistoryUrl = getJobHistoryUrl(applicationId);
             logger.info("jobHistoryUrl={}", jobHistoryUrl);
             responseContent = HttpUtils.get(jobHistoryUrl);
-            JSONObject jsonObject = JSONObject.parseObject(responseContent);
-            result = jsonObject.getJSONObject("job").getString("state");
+            ObjectNode jsonObject = JSONUtils.parseObject(responseContent);
+            result = jsonObject.path("job").path("state").asText();
```

Comment:
result = jsonObject.path("job").path("state").asText() always assign a value to result ,which empty string indicating a null value . In original logic ,null value will throw NPE in the following switch. Now the result is empty string ,then getApplicationStatus method will return ExecutionStatus.RUNNING_EXEUTION. I think you'd better optimize the switch logic